samples/azure-quantum/CONTRIBUTING.md

```diff
@@ -0,0 +1,40 @@
+# Process for creating notebook samples for the Azure Quantum portal experience
+This document focuses on the process for creating samples that will eventually appear in the hosted notebooks sample gallery in the portal. If you are creating a sample for other purposes or means of consumption, then this document might not apply to your scenario. (If you are interested in contributing an optimization notebook sample, please see [this repository](https://github.com/microsoft/qio-samples)).
+
+There are three steps involved in creating a new sample.
+
+1. Write the notebook sample
+1. Test it in the portal
+1. Merge it into `main` AND into `feature/samples-gallery`
+
+Once it is merged, a service engineer must be notified to add it to the sample gallery. Once done, the sample will be publicly available in the next deployment, usually within a week.
+
+## Writing the sample
+There are a few principles to keep in mind when writing samples that are going to be presented in the portal's sample gallery.
+1. The command to log in to Azure Quantum should be in its own cell.
+1. There should be no empty cells in the notebook.
+1. The notebook must be entirely self-contained; it cannot reference other files on disk.
+
+## Testing the sample
+Once the sample is written and working locally, you can test it out in the hosted environment in the Azure portal. In order to make it appear in the sample gallery, you must access the portal using a special link:
+
+`https://portal.azure.com/?microsoft_azure_quantum_include_notebook_sample_with_url={URL of the raw sample file}`.
+Your browser should automatically  URL-encode the URL you paste, so don't worry about special characters.
+
+Make sure that the sample is rendering correctly, all cells run successfully, and the login details are auto-populated with the correct information for the workspace.
+
+## Creating the PRs
+First, create a PR against `main` with the notebook sample. Once that is merged, you are ready to merge it back into `feature/samples-gallery`. You might not be able to merge the `main` branch entirely into `feature/samples-gallery` due to some divergence they have, so just create a new branch based on `feature/samples-gallery` and check out your new files for the notebook samples from `main`. In this PR, you must include a few details:
```

Comment:
Cherry-picking may also be a solution here.

Reply:
True, but I'll keep the document simple and just mention one option.